### PR TITLE
feat: show current dir name in tab

### PR DESCRIPTION
**Pull request by g-plane**
This PR introduces a new option `max_width` in `[tab]` section for controlling the max width of current directory name in tab. Default value is `0` which means directory name won't be displayed in tab. This matches current behavior.

Here is an example for setting value to `10`:

image

The first tab's directory is "raffia" whose length is less than the threshold we set; the second tab's directory is "swc-plugin-react-remove-properties" whose length is greater than the threshold, so it's truncated and only the first 10 chars are displayed. (Should we add ellipses for this case?)

### CODE REVIEW

File: core/src/manager/tab.rs [hunk 1]
```diff
@@ -258,6 +258,10 @@ impl Tab {
 		};
 		true
 	}
+
+	pub fn current_name(&self) -> &str {
+		self.current.cwd.file_name().and_then(|name| name.to_str()).unwrap_or_default()
```
Comment (sxyazi) [inline core/src/manager/tab.rs:263]: Using the `cwd` itself instead of the default might be better when can't get the filename.

File: app/src/header/tabs.rs [hunk 2]
```diff
@@ -19,11 +19,18 @@ impl<'a> Widget for Tabs<'a> {
 			tabs
 				.iter()
 				.enumerate()
-				.map(|(i, _)| {
+				.map(|(i, tab)| {
+					let mut text = format!(" {} ", i + 1);
+					if THEME.tab.max_width > 0 {
+						text
+							.push_str(&tab.current_name().chars().take(THEME.tab.max_width).collect::<String>());
```
Comment (sxyazi) [inline app/src/header/tabs.rs:26]: The number of chars and the display width are different, one Unicode char may take up 2 cells. it's better to use the `unicode-width` crate here.

File: config/preset/theme.toml [hunk 3]
```diff
@@ -1,6 +1,7 @@
 [tab]
-active   = { fg = "#1E2031", bg = "#80AEFA" }
-inactive = { fg = "#C8D3F8", bg = "#484D66" }
+active    = { fg = "#1E2031", bg = "#80AEFA" }
+inactive  = { fg = "#C8D3F8", bg = "#484D66" }
+max_width = 0
```
Comment (sxyazi) [inline config/preset/theme.toml:4]: I think `max_width` should represent the overall width of the tab (with default 1), not just the dir name, since it seems not related to the dir name by looking

**Review by sxyazi — CHANGES_REQUESTED** [review]

File: config/preset/theme.toml (hunk 3 above)
Comment (g-plane) [reply config/preset/theme.toml:4]: I agree. Do you have any suggestions about the naming?

Comment (sxyazi) [reply config/preset/theme.toml:4]: I prefer to keep this name, but change the behavior to fit it, so that it can cover what is added in the future not only the dir name currently.

File: core/src/manager/tab.rs (hunk 1 above)
Comment (g-plane) [reply core/src/manager/tab.rs:263]: Do you mean return `self.current.cwd`?

Comment (sxyazi) [reply core/src/manager/tab.rs:263]: Like this https://github.com/sxyazi/yazi/blob/866ed1704459b91b7e6ea220e4aadcbd1397ba6d/config/src/pattern.rs#L23

File: app/src/header/tabs.rs [hunk 4]
```diff
@@ -19,11 +22,33 @@ impl<'a> Widget for Tabs<'a> {
...
 				.iter()
 				.enumerate()
-				.map(|(i, _)| {
+				.map(|(i, tab)| {
+					let mut text = format!(" {} ", i + 1);
+					if let Some(dir_name) = tab.current_name() {
+						text.push_str(dir_name);
+					}
+
+					let threshold = THEME.tab.max_width.max(3) - 1;
```
Comment (sxyazi) [inline app/src/header/tabs.rs:31]: It seems that `threshold` doesn't need to contain 2 spaces, as the spaces are `padding`, not `content`.

Currently, the spaces should always be present, and the user can be allowed to configure their own padding characters later on.

File: config/preset/theme.toml [hunk 5]
```diff
@@ -1,6 +1,7 @@
 [tab]
-active   = { fg = "#1E2031", bg = "#80AEFA" }
-inactive = { fg = "#C8D3F8", bg = "#484D66" }
+active    = { fg = "#1E2031", bg = "#80AEFA" }
+inactive  = { fg = "#C8D3F8", bg = "#484D66" }
+max_width = 3
```
Comment (sxyazi) [inline config/preset/theme.toml:4]: Should be set to `1`, i.e. `padding` is not included.

**Review by sxyazi — CHANGES_REQUESTED** [review]

**Comment by sxyazi** [comment]
Thanks for your work, merged!
